src/Test/Digest.hs

```diff
@@ -32,20 +37,25 @@ main = testBuild test $ do
         forM_ outs $ \out -> writeFile' out "rewrite"
 
 
-test build = do
+test :: IORef (HashMap FilePath FileHash) -> ([String] -> IO ()) -> IO ()
+test ref build = do
     let outs = ["Out.txt","Out1.txt","Out2.txt"]
     let writeOut x = forM_ outs $ \out -> writeFile out x
     let writeIn = writeFile "In.txt"
     let assertOut x = forM_ outs $ \out -> assertContents out x
+    let clearTrackedHashes = atomicModifyIORef' ref $ \_ -> (HashMap.empty, ())
+    let getTrackedHashes  = atomicModifyIORef' ref $ \x -> (x, x)
+    let assertEmptyHashes = assertNoHashes ref
 
     writeOut ""
     writeIn "X"
     build ["--sleep","--digest-and"]
     assertOut "X"
 
-    -- should not involve a hash calculation (sadly no way to test that)
+    clearTrackedHashes
     build ["--sleep","--digest-and"]
     assertOut "X"
+    assertEmptyHashes
```

Comment:
So this is the failing step? The comment says there should be no hash, but you are saying this does perform a hash?

Reply:
Yes that's right, the test fails here for me.